docs/docs/how-tos/domain-registry.md

```diff
@@ -6,12 +6,81 @@ description: Setting a custom DNS Domain Registry
 
 # Nebari domain registry setup
 
-This domain will be where your application will be exposed.
+## What is a DNS?
 
-Currently, Nebari's default configuration is set for CloudFlare for automatic DNS registration. If you prefer an alternate DNS provider, you'll need to change the `--dns-provider` flag from `cloudflare` to `none` on the QHub deploy command.
-The deployment process will then pause when it asks for an IP address (or `CNAME`, if using AWS) and prompt you to provide the desired URL.
+The Domain Name System (DNS) turns domain names into IP addresses, which browsers use to load internet pages.
+Every device connected to the internet has its IP address, which other devices use to locate the device.
+DNS name servers are used to locate the IP address of a domain name using words or instead of direct IP addresses.
+
+## Setting up a DNS
+
+During deployment, Qhub will generate an [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) IP for connection with the Kubernetes cluster and all related services that Qhub runs. If not automatically handled, Qhub will request the user to generate the necessary [DNS records](https://www.cloudflare.com/en-gb/learning/dns/dns-records/) and update the domain within the newly created IP:
+
+```bash
+Take IP Address 12.312.312.312 and update DNS to point to "your.domain" [Press Enter when Complete]
+```
+
+Once the IP is generated, you will need to grab it and create the necessary records within the DNS provider of your choice.
+
+## Cloudflare
+
+QHub supports Cloudflare as a DNS provider. If you choose to use Cloudflare, first create an account, then there are two possible following options:
+
+1. You can register your application domain name on it, using the [Cloudflare nameserver](https://developers.cloudflare.com/dns/zone-setups/full-setup/setup) (recommended).
+
+2. You can outright buy a new domain with Cloudflare (this action isn’t particularly recommended).
+
+To generate a token [follow these steps](https://developers.cloudflare.com/api/tokens/create).
```

Comment:
I'm confused. Am I supposed to click on the linked words "follow these steps" or am I supposed to follow the steps below?

This is what I think you mean:

```suggestion
To generate a token follow the steps below. For additional information, see the [CloudFlare docs](https://developers.cloudflare.com/api/tokens/create).
```